src/main/kotlin/no/nav/familie/ef/sak/mapper/AleneomsorgMapper.kt

```diff
@@ -0,0 +1,90 @@
+package no.nav.familie.ef.sak.mapper
+
+import no.nav.familie.ef.sak.api.gui.dto.*
+import no.nav.familie.ef.sak.integration.dto.pdl.PdlAnnenForelder
+import no.nav.familie.ef.sak.integration.dto.pdl.PdlBarn
+import no.nav.familie.ef.sak.integration.dto.pdl.PdlSøker
+import no.nav.familie.ef.sak.integration.dto.pdl.visningsnavn
+import no.nav.familie.kontrakter.ef.søknad.SøknadOvergangsstønad as Søknad
+
+
+object AleneomsorgMapper {
+
+    fun tilDto(pdlSøker: PdlSøker,
+               pdlBarn: Map<String, PdlBarn>,
+               barneforeldre: Map<String, PdlAnnenForelder>,
+               søknad: Søknad): Aleneomsorg {
+
+        val søkersFnr = søknad.personalia.verdi.fødselsnummer.verdi.verdi
+        val alleBarn = BarnMatcher.kobleSøknadsbarnOgRegisterBarn(søknad.barn.verdi, pdlBarn)
+
+        return Aleneomsorg(alleBarn.map { tilBarn(it, barneforeldre, søkersFnr) },
+                           pdlSøker.bostedsadresse.map { Adresse(it) })
+
+    }
+
+    private fun tilBarn(barn: MatchetBarn,
+                        barneforeldre: Map<String, PdlAnnenForelder>,
+                        søkersFnr: String): Barn {
+        val navn = barn.pdlBarn?.navn?.firstOrNull()?.visningsnavn() ?: barn.søknadsbarn.navn?.verdi ?: "Ubestemt"
+        val fødselsnummer = barn.fødselsnummer ?: barn.søknadsbarn.fødselsnummer?.verdi?.verdi
+        val termindatoFødselsdato = barn.pdlBarn?.fødsel?.firstOrNull()?.fødselsdato ?: barn.søknadsbarn.fødselTermindato?.verdi
+        val begrunnelseIkkeOppgittAnnenForelder =
+                barn.søknadsbarn.annenForelder?.verdi?.ikkeOppgittAnnenForelderBegrunnelse?.verdi
+        val annenForelder = tilAnnenForelder(barn, barneforeldre, søkersFnr)
+        val skalBoBorHosSøker = barn.søknadsbarn.harSkalHaSammeAdresse.verdi
+        val deltBosted = tilDeltBosted(barn)
+        val fraRegister = barn.pdlBarn != null
+
+        return Barn(navn,
+                    fødselsnummer,
+                    termindatoFødselsdato,
+                    begrunnelseIkkeOppgittAnnenForelder,
+                    annenForelder,
+                    skalBoBorHosSøker,
+                    deltBosted,
+                    fraRegister)
+    }
+
+    private fun tilAnnenForelder(barn: MatchetBarn, barneforeldre: Map<String, PdlAnnenForelder>, søkersFnr: String): Forelder? {
+
+        val fnr = (barn.pdlBarn?.familierelasjoner?.firstOrNull { it.relatertPersonsIdent != søkersFnr }?.relatertPersonsIdent
+                   ?: barn.søknadsbarn.annenForelder?.verdi?.person?.verdi?.fødselsnummer?.verdi?.verdi)
+
+        val pdlAnnenForelder = barneforeldre[fnr]
+
+        val bostedsland = if (pdlAnnenForelder?.bostedsadresse != null) {
+            "NORGE"
```

Comment:
Tror ikke dette er helt vanntett, se https://navikt.github.io/pdl/#_hvordan_vite_om_en_person_er_bosatt_i_norge_eller_i_utlandet 

> Man kan ha bostedsadresse, oppholdsadresse og kontaktadresse. Alle disse adressetypene kan være enten norske eller utenlandske. Når man har en norsk bostedsadresse vil den alltid komme fra folkeregisteret. Utenlandske bostedsadresser vil vi registrere dersom de er bekreftet av utenlandske myndigheter.

Reply:
For å regnes som bosatt i Norge må man ha norsk bostedsadresse. Hvis du ikke har det, så er du ikke bosatt i Norge. Linken sier jo riktignok at folkerregisterloven og folketrygdloven sier at det er forskjell på hvordan de lovene tolker bosatt, men uten egentlig å gi noe svar på det. 
Tror det kan være letter for arbeidet videre og ta denne branchen som den er, og opprette en oppgave på å unddersøke det nærmere.

Reply:
Ja, så du mangler vel en sjekk på at bostedsadressen er norsk. Lag gjerne en oppgave på det, enig i at det bør diskuteres med fag. Du kan også utvide sjekken til å se om det finnes enten vegadresse eller matrikkeladresse på bostedsadressen - i så fall må den være norsk da utenlandske adresser er en egen type.

Reply:
Bostedsadresse er alltid norsk, men jeg ser at de planlegger å innføre utenlandske bostedsadrresser. Tror vvi må se an hvordan dette blir formattert fra PDL sin side.

Reply:
Det er nettopp det den ikke er i følge dokumentasjonen... Slik jeg tolker teksten er utenlandskadresse en adressetype som legges under bostedsadresse på samme måte som vegadresse og matrikkeladresse. Legg i det minste en TODO i koden på det.

Reply:
Fikset nå.